lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/runner/NativeExecCommand.scala

```diff
@@ -26,7 +29,22 @@ object NativeExecCommand {
     val fullExecPath =
       dm.paths.engines.resolve(version).resolve("bin").resolve(execName)
 
-    if (fullExecPath.toFile.exists()) Some(NativeExecCommand(fullExecPath))
-    else None
+    if (fullExecPath.toFile.exists() && isBinary(fullExecPath)) {
+      Some(NativeExecCommand(fullExecPath))
+    } else None
+  }
+
+  private def isBinary(path: Path): Boolean = {
```

Comment:
Alternative check would be:
```
return !Files.readLines("bin/enso").get(0).startsWith("#!/bin/sh");
```
that is a common prefix of most of the shell scripts. However we'd have to add such header to our shell script:
```diff
enso$ git diff
diff --git distribution/bin/enso distribution/bin/enso
index 1939fc99dc..93a45f4a6e 100755
--- distribution/bin/enso
+++ distribution/bin/enso
@@ -1,3 +1,5 @@
+#!/bin/sh
+
 COMP_PATH=$(dirname "$0")/../component
 
 JAVA_OPTS="--add-opens=java.base/java.nio=ALL-UNNAMED $JAVA_OPTS"
```
but as you say, Tika was already available as it is used somewhere.

Reply:
That wouldn't work on Windows

Reply:
But on windows the difference is in extension:
- `enso.bat` - is the shell script
- `enso.exe` - is the native binary

We only need the content file check for Unixes.